src/types/tags.ts

```diff
@@ -1,7 +1,15 @@
+export const unset = Symbol('unset')
+export type Unset = typeof unset
+
 export type QueryTag<
- TName extends string = string,
+ TData extends unknown = Unset,
 > = {
```

Comment:
This `Unset` type is important to call out. Originally I had the default data type for a tag as `unknown`, however that caused issues where if you added an untyped tag (with type `unknown`) and a typed tag that was an incorrect type, typescript would not error because it would widen `[tag(), tag<string>()` to `QueryTag<unknown>[]`. You could get the correct type error by using `as const` but I wouldn't expect users to know, remember, or care to do that.

Making the default type for a tag something highly specific (a symbol) means that typescript accurately reports errors. It will make our internal types more complex but it produces the desired devx for users.